### PR TITLE
Allow a failover payload to pass validations

### PR DESCRIPTION
**Pull request by rtorrero**
This PR fixes an issue introduced in PR #602 that caused a payload validation failure in the event of a failover. This caused that some information in the SAP systems overview and HANA database details overview was wrong/missing.

Now both views show the correct information:
sap-details
db-details

### CODE REVIEW

File: lib/trento/application/integration/discovery/payloads/sap_system_discovery_payload.ex
Comment (arbulu89) [inline lib/trento/application/integration/discovery/payloads/sap_system_discovery_payload.ex:279]: I don't really understand this. Why are you removing these fields? They should be always coming.
Do they have some empty field, and that's why ecto is discarding the messages?

What are the values that you are receiving in the failed message?

File: lib/trento/application/integration/discovery/payloads/sap_system_discovery_payload.ex
Comment (arbulu89) [inline lib/trento/application/integration/discovery/payloads/sap_system_discovery_payload.ex:344]: We should rename the methods adding a `maybe_` here, as this methods doesn't validate in all the cases

**Review by arbulu89 — COMMENTED** [review]
Thanks @rtorrero ,
I think we should add a new test (unit test) at least on this, using the payload which caused the issue. This way we continue covering the most standard use-cases with tests, so we know that we don't break them

File: lib/trento/application/integration/discovery/payloads/sap_system_discovery_payload.ex
Comment (rtorrero) [reply lib/trento/application/integration/discovery/payloads/sap_system_discovery_payload.ex:279]: Forn @abravosuse to trigger the failover scenario for me, he basically was stopping the primary instance; causing that the process information in the payload was something like:

```
"Processes" => [ % {
	"description" => "HDB Daemon",
	"dispstatus" => "SAPControl-GRAY",
	"name" => "hdbdaemon",
	"pid" => 26205,
	"textstatus" => "Stopped"
}],
```

As you see, that process is in status Stopped and doesn't include these fields.

Comment (arbulu89) [reply lib/trento/application/integration/discovery/payloads/sap_system_discovery_payload.ex:279]: But this is already the elixir code. I would like to see the content of the real payload, to understand why is not casting those fields. Maybe there are not coming. And that's why having a test with the real payload would be ideal

**Review by dottorblaster — CHANGES_REQUESTED** [review]
As Xabi said, we have to add this case to the unit tests to be sure we don't end up in regressions :+1: 

Good job! Once we are done with this we can merge

**Review by dottorblaster — APPROVED** [review]
Now that we've got the test for the failing case, LGTM

**Review by arbulu89 — APPROVED** [review]
Thanks for adding the test @rtorrero 

In my opinion, we shouldn't call this tests and fixture "failover". The failover is a cluster thing, not a sap system thing. In this case, we are just watching a stopped SAP instance, which launches a failover in the cluster (making the other secondary instance primary).
I would try to be meticulous with the wording, and call this scenarios as `stopped` rather than `failover`.

File: lib/trento/application/integration/discovery/payloads/sap_system_discovery_payload.ex
Comment (arbulu89) [reply lib/trento/application/integration/discovery/payloads/sap_system_discovery_payload.ex:279]: Thanks @rtorrero

**Comment by rtorrero** [comment]
Added the `maybe_` prefix for the validation functions and changed a bit the wording for the test case.

**Pull request merged**
